File: lib/src/modules/seller_product/models/create_sku_request.dart
class CreateSkuRequest {
  final String skuCode;
  final double price;
  final int stock;
  Map<String, String> specifications;

  CreateSkuRequest({
    required this.skuCode,
    required this.price,
    required this.stock,
    required this.specifications,
  });

  Map<String, dynamic> toJson() {
    return {
      'skuCode': skuCode,
      'price': price,
      'stock': stock,
      'specifications': specifications,
    };
  }
}

File: src/main/java/com/example/sneakerx/dtos/shop/CreateSkuRequest.java
package com.example.sneakerx.dtos.shop;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Map;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CreateSkuRequest {
    private String skuCode;
    private Double price;
    private Integer stock;

    Map<String, String> specifications;
}
